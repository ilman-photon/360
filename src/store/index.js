import { createSlice } from "@reduxjs/toolkit";

const INITIAL_STATE = {
  loading: true,
  counter: 0,
};

export const indexStore = createSlice({
  name: "index",
  initialState: INITIAL_STATE,
  reducers: {
    setLoading: (state, action) => {
      state.loading = action.payload;
    },
    increment: (state, action) => {
      state.counter++;
    },
  },
});

// Action creators are generated for each case reducer function
export const { setLoading, increment } = indexStore.actions;

export default indexStore.reducer;
