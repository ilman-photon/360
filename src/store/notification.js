import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";

export const fetchNotifications = createAsyncThunk(
  "user/fetchNotifications",
  async () => {
    return fetch("/api/dummy/notification").then((res) => res.json());
  }
);

const INITIAL_STATE = {
  list: [],
  status: "loading",
};

export const notificationStore = createSlice({
  name: "index",
  initialState: INITIAL_STATE,
  reducers: {
    markAllAsRead: (state) => {
      state.list = state.list.map((v) => {
        return {
          ...v,
          isRead: true,
        };
      });
    },
  },
  extraReducers: {
    [fetchNotifications.pending]: (state) => {
      state.status = "loading";
    },
    [fetchNotifications.fulfilled]: (state, { payload }) => {
      state.list = payload;
      state.status = "success";
    },
    [fetchNotifications.rejected]: (state) => {
      state.status = "failed";
    },
  },
});

// Action creators are generated for each case reducer function
export const { markAllAsRead } = notificationStore.actions;

export default notificationStore.reducer;
