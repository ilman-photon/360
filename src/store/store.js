import { configureStore } from "@reduxjs/toolkit";
import index from "../store/index";
import user from "../store/user";

export const store = configureStore({
  reducer: {
    user,
    index,
  },
});

export default store;
