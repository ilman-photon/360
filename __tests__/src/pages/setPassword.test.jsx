import { act, fireEvent, render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import store from "../../../src/store/store";
import SetPasswordPage, { getServerSideProps } from "../../../src/pages/patient/set-password";

describe("Set Password", () => {
  let container;
  beforeEach(async () => {
    const contex = {
      query: {
        username: "smith1@photon.com"
      },
    };
    getServerSideProps(contex);
    container = render(
      <Provider store={store}>
        {SetPasswordPage.getLayout(<SetPasswordPage />)}
      </Provider>
    );
    await waitFor(() => {
      container.getByText(/Set Password/i)
    })
  });
  test("renders forgot password", () => {
    expect(container).toMatchSnapshot();
  });

  test("check valid email", () => {
    setTimeout(() => {
      const title = container.getByText(/title/i);
      expect("title").toEqual(title.textContent);
    }, 500);
  });

  test("click reset password", async () => {
    // setTimeout(() => {
    await waitFor(() => {
      container.getByText(/Set Password/i)
    })
    //const password = container.getById(/password/i);
    const password = container.container.querySelector('#password')
    fireEvent.change(password, { target: { value: "user123@A" } });
    expect(password.value).toEqual("user123@A");

    // const confirmPassword = container.getById(
    //   /confirmPassword/i
    // );
    const confirmPassword = container.container.querySelector('#confirmPassword')
    fireEvent.change(confirmPassword, { target: { value: "user123@A" } });
    expect(confirmPassword.value).toEqual("user123@A");
    // }, 500);

    const submitbutton = container.getByRole("button", {name: /Create Account/i})
    fireEvent.click(submitbutton)
  });
});
