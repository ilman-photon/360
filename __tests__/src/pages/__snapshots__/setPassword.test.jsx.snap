// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Set Password renders forgot password 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="contanierPage MuiBox-root css-0"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root container css-2q77k0-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            style="display: flex; flex-direction: column; padding: 0px; margin-top: 0px;"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
            >
              Set Password
            </h2>
            <h4
              class="MuiTypography-root MuiTypography-h4 css-sybkak-MuiTypography-root"
            >
              Enter a password to setup your account.
            </h4>
            <div
              style="margin: 8px;"
            />
            <form
              style="display: grid; margin-top: 8px;"
            >
              <div
                class="MuiFormControl-root MuiTextField-root custom-input css-1aweqgj-MuiFormControl-root-MuiTextField-root"
                style="margin-top: 11px;"
                withicon="true"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-disabled css-pl8awd-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="username"
                  id="username-label"
                >
                  Username
                </label>
                <div
                  class="MuiFilledInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-sizeSmall css-jq7ton-MuiInputBase-root-MuiFilledInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiFilledInput-input MuiInputBase-input Mui-disabled MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
                    disabled=""
                    id="username"
                    placeholder=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="MuiFormControl-root css-1c2vtuz-MuiFormControl-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-xnwbhr-MuiFormControl-root-MuiTextField-root"
                  customevent="[object Object]"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-1suhfmo-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="password"
                    id="password-label"
                  >
                    Password
                  </label>
                  <div
                    class="MuiFilledInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-v0xomr-MuiInputBase-root-MuiFilledInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1gctnaj-MuiInputBase-input-MuiFilledInput-input"
                      id="password"
                      placeholder=""
                      type="password"
                      value=""
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-jv61v3-MuiInputAdornment-root"
                    >
                      <button
                        aria-label="toggle password visibility"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="VisibilityOutlinedIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 6c3.79 0 7.17 2.13 8.82 5.5C19.17 14.87 15.79 17 12 17s-7.17-2.13-8.82-5.5C4.83 8.13 8.21 6 12 6m0-2C7 4 2.73 7.11 1 11.5 2.73 15.89 7 19 12 19s9.27-3.11 11-7.5C21.27 7.11 17 4 12 4zm0 5c1.38 0 2.5 1.12 2.5 2.5S13.38 14 12 14s-2.5-1.12-2.5-2.5S10.62 9 12 9m0-2c-2.48 0-4.5 2.02-4.5 4.5S9.52 16 12 16s4.5-2.02 4.5-4.5S14.48 7 12 7z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                <div
                  class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                  style="min-height: 0px;"
                >
                  <div
                    class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                  >
                    <div
                      class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                    >
                      <div
                        class="MuiBox-root css-0"
                        style="display: flex; flex-direction: row; margin-top: 15px;"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-k5i2wh-MuiSvgIcon-root"
                          data-testid="CancelIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                          />
                        </svg>
                        <span
                          style="margin-left: 10px; color: rgb(8, 7, 7); font-family: Libre Franklin; font-style: normal; font-weight: 400; font-size: 14px; line-height: 18px;"
                        >
                          Length: 8-20 characters
                        </span>
                      </div>
                      <div
                        class="MuiBox-root css-0"
                      >
                        <div
                          class="MuiBox-root css-0"
                          style="display: flex; flex-direction: row; margin-top: 15px;"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-k5i2wh-MuiSvgIcon-root"
                            data-testid="CancelIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                            />
                          </svg>
                          <span
                            style="margin-left: 10px; color: rgb(8, 7, 7); font-family: Libre Franklin; font-style: normal; font-weight: 400; font-size: 14px; line-height: 18px;"
                          >
                            Contain at least 3 out of 4 types of characters below:
                          </span>
                        </div>
                        <ul>
                          <li
                            style="color: rgb(8, 7, 7);"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body1 css-1319ain-MuiTypography-root"
                            >
                              At least One Numeric
                            </p>
                          </li>
                          <li
                            style="color: rgb(8, 7, 7);"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body1 css-1319ain-MuiTypography-root"
                            >
                              At least One Upper case Alpha
                            </p>
                          </li>
                          <li
                            style="color: rgb(8, 7, 7);"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body1 css-1319ain-MuiTypography-root"
                            >
                              At least One Lower case Alpha
                            </p>
                          </li>
                          <li
                            style="color: rgb(8, 7, 7);"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body1 css-1319ain-MuiTypography-root"
                            >
                              At least One Special character (no spaces)
                            </p>
                          </li>
                        </ul>
                      </div>
                      <div
                        class="MuiBox-root css-0"
                        style="display: flex; flex-direction: row; margin-top: 15px;"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-k5i2wh-MuiSvgIcon-root"
                          data-testid="CancelIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                          />
                        </svg>
                        <span
                          style="margin-left: 10px; color: rgb(8, 7, 7); font-family: Libre Franklin; font-style: normal; font-weight: 400; font-size: 14px; line-height: 18px;"
                        >
                          Password should not contain your username
                        </span>
                      </div>
                      <div
                        class="MuiBox-root css-0"
                        style="display: flex; flex-direction: row; margin-top: 15px;"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-k5i2wh-MuiSvgIcon-root"
                          data-testid="CancelIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                          />
                        </svg>
                        <span
                          style="margin-left: 10px; color: rgb(8, 7, 7); font-family: Libre Franklin; font-style: normal; font-weight: 400; font-size: 14px; line-height: 18px;"
                        >
                          New password must not match current password
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiFormControl-root css-1c2vtuz-MuiFormControl-root"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-xnwbhr-MuiFormControl-root-MuiTextField-root"
                  customevent="[object Object]"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-1suhfmo-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="confirmPassword"
                    id="confirmPassword-label"
                  >
                    Confirm Password
                  </label>
                  <div
                    class="MuiFilledInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-v0xomr-MuiInputBase-root-MuiFilledInput-root"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1gctnaj-MuiInputBase-input-MuiFilledInput-input"
                      id="confirmPassword"
                      placeholder=""
                      type="password"
                      value=""
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-jv61v3-MuiInputAdornment-root"
                    >
                      <button
                        aria-label="toggle password visibility"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="VisibilityOutlinedIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 6c3.79 0 7.17 2.13 8.82 5.5C19.17 14.87 15.79 17 12 17s-7.17-2.13-8.82-5.5C4.83 8.13 8.21 6 12 6m0-2C7 4 2.73 7.11 1 11.5 2.73 15.89 7 19 12 19s9.27-3.11 11-7.5C21.27 7.11 17 4 12 4zm0 5c1.38 0 2.5 1.12 2.5 2.5S13.38 14 12 14s-2.5-1.12-2.5-2.5S10.62 9 12 9m0-2c-2.48 0-4.5 2.02-4.5 4.5S9.52 16 12 16s4.5-2.02 4.5-4.5S14.48 7 12 7z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <button
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root customButton customButtonLarge  css-i2fzad-MuiButtonBase-root-MuiButton-root"
                style="margin-top: 16px; margin-left: 0px; margin-right: 0px; text-align: center;"
                tabindex="0"
                type="submit"
              >
                Create Account
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </form>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-160akwi-MuiTypography-root-MuiLink-root"
              style="margin-top: 16px; margin-left: 0px; margin-right: 0px; text-align: center; cursor: pointer;"
            >
              backButtonLink
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="contanierPage MuiBox-root css-0"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root container css-2q77k0-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          style="display: flex; flex-direction: column; padding: 0px; margin-top: 0px;"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
          >
            Set Password
          </h2>
          <h4
            class="MuiTypography-root MuiTypography-h4 css-sybkak-MuiTypography-root"
          >
            Enter a password to setup your account.
          </h4>
          <div
            style="margin: 8px;"
          />
          <form
            style="display: grid; margin-top: 8px;"
          >
            <div
              class="MuiFormControl-root MuiTextField-root custom-input css-1aweqgj-MuiFormControl-root-MuiTextField-root"
              style="margin-top: 11px;"
              withicon="true"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-disabled css-pl8awd-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="username"
                id="username-label"
              >
                Username
              </label>
              <div
                class="MuiFilledInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-sizeSmall css-jq7ton-MuiInputBase-root-MuiFilledInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiFilledInput-input MuiInputBase-input Mui-disabled MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
                  disabled=""
                  id="username"
                  placeholder=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiFormControl-root css-1c2vtuz-MuiFormControl-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-xnwbhr-MuiFormControl-root-MuiTextField-root"
                customevent="[object Object]"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-1suhfmo-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="password"
                  id="password-label"
                >
                  Password
                </label>
                <div
                  class="MuiFilledInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-v0xomr-MuiInputBase-root-MuiFilledInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1gctnaj-MuiInputBase-input-MuiFilledInput-input"
                    id="password"
                    placeholder=""
                    type="password"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-jv61v3-MuiInputAdornment-root"
                  >
                    <button
                      aria-label="toggle password visibility"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="VisibilityOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 6c3.79 0 7.17 2.13 8.82 5.5C19.17 14.87 15.79 17 12 17s-7.17-2.13-8.82-5.5C4.83 8.13 8.21 6 12 6m0-2C7 4 2.73 7.11 1 11.5 2.73 15.89 7 19 12 19s9.27-3.11 11-7.5C21.27 7.11 17 4 12 4zm0 5c1.38 0 2.5 1.12 2.5 2.5S13.38 14 12 14s-2.5-1.12-2.5-2.5S10.62 9 12 9m0-2c-2.48 0-4.5 2.02-4.5 4.5S9.52 16 12 16s4.5-2.02 4.5-4.5S14.48 7 12 7z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div
                class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
                style="min-height: 0px;"
              >
                <div
                  class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                  >
                    <div
                      class="MuiBox-root css-0"
                      style="display: flex; flex-direction: row; margin-top: 15px;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-k5i2wh-MuiSvgIcon-root"
                        data-testid="CancelIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                        />
                      </svg>
                      <span
                        style="margin-left: 10px; color: rgb(8, 7, 7); font-family: Libre Franklin; font-style: normal; font-weight: 400; font-size: 14px; line-height: 18px;"
                      >
                        Length: 8-20 characters
                      </span>
                    </div>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <div
                        class="MuiBox-root css-0"
                        style="display: flex; flex-direction: row; margin-top: 15px;"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-k5i2wh-MuiSvgIcon-root"
                          data-testid="CancelIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                          />
                        </svg>
                        <span
                          style="margin-left: 10px; color: rgb(8, 7, 7); font-family: Libre Franklin; font-style: normal; font-weight: 400; font-size: 14px; line-height: 18px;"
                        >
                          Contain at least 3 out of 4 types of characters below:
                        </span>
                      </div>
                      <ul>
                        <li
                          style="color: rgb(8, 7, 7);"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-1319ain-MuiTypography-root"
                          >
                            At least One Numeric
                          </p>
                        </li>
                        <li
                          style="color: rgb(8, 7, 7);"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-1319ain-MuiTypography-root"
                          >
                            At least One Upper case Alpha
                          </p>
                        </li>
                        <li
                          style="color: rgb(8, 7, 7);"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-1319ain-MuiTypography-root"
                          >
                            At least One Lower case Alpha
                          </p>
                        </li>
                        <li
                          style="color: rgb(8, 7, 7);"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-1319ain-MuiTypography-root"
                          >
                            At least One Special character (no spaces)
                          </p>
                        </li>
                      </ul>
                    </div>
                    <div
                      class="MuiBox-root css-0"
                      style="display: flex; flex-direction: row; margin-top: 15px;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-k5i2wh-MuiSvgIcon-root"
                        data-testid="CancelIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                        />
                      </svg>
                      <span
                        style="margin-left: 10px; color: rgb(8, 7, 7); font-family: Libre Franklin; font-style: normal; font-weight: 400; font-size: 14px; line-height: 18px;"
                      >
                        Password should not contain your username
                      </span>
                    </div>
                    <div
                      class="MuiBox-root css-0"
                      style="display: flex; flex-direction: row; margin-top: 15px;"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-k5i2wh-MuiSvgIcon-root"
                        data-testid="CancelIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                        />
                      </svg>
                      <span
                        style="margin-left: 10px; color: rgb(8, 7, 7); font-family: Libre Franklin; font-style: normal; font-weight: 400; font-size: 14px; line-height: 18px;"
                      >
                        New password must not match current password
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiFormControl-root css-1c2vtuz-MuiFormControl-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-xnwbhr-MuiFormControl-root-MuiTextField-root"
                customevent="[object Object]"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-1suhfmo-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="confirmPassword"
                  id="confirmPassword-label"
                >
                  Confirm Password
                </label>
                <div
                  class="MuiFilledInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-v0xomr-MuiInputBase-root-MuiFilledInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1gctnaj-MuiInputBase-input-MuiFilledInput-input"
                    id="confirmPassword"
                    placeholder=""
                    type="password"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-jv61v3-MuiInputAdornment-root"
                  >
                    <button
                      aria-label="toggle password visibility"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="VisibilityOutlinedIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 6c3.79 0 7.17 2.13 8.82 5.5C19.17 14.87 15.79 17 12 17s-7.17-2.13-8.82-5.5C4.83 8.13 8.21 6 12 6m0-2C7 4 2.73 7.11 1 11.5 2.73 15.89 7 19 12 19s9.27-3.11 11-7.5C21.27 7.11 17 4 12 4zm0 5c1.38 0 2.5 1.12 2.5 2.5S13.38 14 12 14s-2.5-1.12-2.5-2.5S10.62 9 12 9m0-2c-2.48 0-4.5 2.02-4.5 4.5S9.52 16 12 16s4.5-2.02 4.5-4.5S14.48 7 12 7z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root customButton customButtonLarge  css-i2fzad-MuiButtonBase-root-MuiButton-root"
              style="margin-top: 16px; margin-left: 0px; margin-right: 0px; text-align: center;"
              tabindex="0"
              type="submit"
            >
              Create Account
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </form>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-160akwi-MuiTypography-root-MuiLink-root"
            style="margin-top: 16px; margin-left: 0px; margin-right: 0px; text-align: center; cursor: pointer;"
          >
            backButtonLink
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
